sark/tseries.py

```diff
@@ -0,0 +1,171 @@
+from pathlib import Path
+from typing import List, TypeVar, Union
+from warnings import warn
+
+import pandas as pd
+
+_time_units = (
+    "days",
+    "day",
+    "hours",
+    "hour",
+    "minutes",
+    "minute",
+    "seconds",
+    "second",
+)
+
+_path_t = Union[str, Path]
+_col_t = TypeVar("_col_t", int, str)
+
+
+def read_timeseries(
+    fpath: _path_t,
+    *,
+    date_cols: Union[List[_col_t], None] = None,
+    col_units: Union[str, None] = None,
+    zero_idx: bool = False,
+    source_t: str = "",
+    **kwargs,
+):
+    """Read a time series from a file.
+
+    While the natural way to structure a time series dataset is with the index
+    column as datetime values, with subsequent columns holding other values,
+    there are a few other frequently used structures.
+
+    The first is to structure it as a table:
+
+    ===========  ===  ===  ====  ====  ====
+                  1    2    ..    23    24
+    ===========  ===  ===  ====  ====  ====
+     1/1/2016      0   10    ..   2.3   5.1
+     4/1/2016      3   11    ..   4.3   9.1
+    ===========  ===  ===  ====  ====  ====
+
+    When `source_t` is set to "table", this function reads a tabular dataset,
+    like the one above, and flattens it into a series, while setting the
+    appropriate datetime values as their index.
+
+    The other common structure is to split the datetime values into multiple
+    columns in the table:
+
+    ===========  ======  ======  ======
+      date        time    col1    col2
+    ===========  ======  ======  ======
+     1/1/2016     10:00    42.0    foo
+     4/1/2016     11:00    3.14    bar
+    ===========  ======  ======  ======
+
+    When `source_t` is set to "multicol". this function the table, while
+    combining the designated columns to construct the datetime values, which
+    are then set as the index.
+
+    If `source_t` is not specified (or set to an empty string), options
+    specific to this function are ignored, and all other keyword options are
+    passed on to the backend transparently; it's a noop.
```

Comment:
a noop?

Reply:
I guess that's jargon.  It's a common contraction for no-operation.  Since the behaviour is to pass the keyword arguments to the backend as-is, with no modifications, it's a "noop".  Do you have a better alternative?

Reply:
Depends on the audience, I suppose. I'd just avoid mentioning that it's a noop (your description didn't come up in the top seach results for the term...). You could just state that all other keyword options are passed onto `pandas.read_csv` transparently, since that allows a user to easily refer to the pandas docs on what kwargs are available?

Reply:
Interestingly, urban dictionary includes [noop](https://www.urbandictionary.com/define.php?term=noop)!  But I take your point, it's best to remove any jargon. Fixed.